packages/server/lib/cloud/artifacts/screenshot_artifact.ts

```diff
@@ -0,0 +1,48 @@
+import fs from 'fs/promises'
+import Debug from 'debug'
+import { sendFile } from '../upload/send_file'
+import { performance } from 'perf_hooks'
+import type { BaseArtifact, ArtifactUploadResult } from './types'
+import { Artifact } from './artifact'
+
+const debug = Debug('cypress:server:cloud:artifacts:screenshot')
+
+export class ScreenshotArtifact extends Artifact implements BaseArtifact {
+  public readonly reportKey = 'screenshots'
+
+  public static async create (filePath, uploadUrl): Promise<ScreenshotArtifact> {
+    const { size: fileSize } = await fs.stat(filePath)
+
+    return new ScreenshotArtifact(filePath, uploadUrl, fileSize)
+  }
+
+  public static createBatch (
+    screenshotUploadUrls: {screenshotId: string, uploadUrl: string}[],
+    screenshotFiles: {screenshotId: string, path: string}[],
+  ): Promise<ScreenshotArtifact>[] {
+    return screenshotUploadUrls.reduce((acc: Promise<ScreenshotArtifact>[], { screenshotId, uploadUrl }) => {
+      const correlatedFilePath = screenshotFiles.find((pathPair) => {
+        return pathPair.screenshotId === screenshotId
+      })?.path
+
+      debug('correlated filepath: %s', correlatedFilePath)
+
+      return correlatedFilePath ?
+        acc.concat(ScreenshotArtifact.create(correlatedFilePath, uploadUrl)) :
+        acc
+    }, [])
+  }
+
+  public async upload (): Promise<ArtifactUploadResult> {
```

Comment:
Not psyched about the repetition here, but not super important to resolve before merge